### DISCUSSION TITLE
[Question] Rom-RB support

### DISCUSSION THREAD

**Discussion by drselump14** [palkan/logidze#197] [Ideas]
Is it possible to integrate this gem to non-AR project. For example: rom-rb

**Comment by palkan** [comment]
Currently, not. But a PR would be welcomed)

Here is a sneak preview of what a refactoring could be like:
- First, we need to extract SQL definitions from AR generators/migrations (not difficult, we already keep them in separate files)
- Secondly, we need to define a value object or type object or whatever, which would play a similar role as `History`  and `History::Type` do for AR.

I'm not a Rom-RB user, so I'm not going to work on this myself.
However, if you can sketch an idea of the desired API, we would be glad to do all the preliminary work (i.e., extract the _core_).

**Reply by drselump14** [reply]
Thanks for the reply and for giving me insight into how to refactor it. I'll try it
